src/Configuration/ConfigurationFactory.php

```diff
@@ -243,14 +240,8 @@ private static function retrieveCoverageBasePath(
 
     private static function retrieveTestFrameworkExtraOptions(
         ?string $testFrameworkExtraOptions,
-        SchemaConfiguration $schema,
-        string $testFramework
-    ): TestFrameworkExtraOptions {
-        $extraOptions = $testFrameworkExtraOptions ?? $schema->getTestFrameworkExtraOptions();
-
-        return $testFramework === TestFrameworkTypes::PHPUNIT
-            ? new PhpUnitExtraOptions($extraOptions)
-            : new PhpSpecExtraOptions($extraOptions)
-        ;
+        SchemaConfiguration $schema
+    ): string {
+        return $testFrameworkExtraOptions ?? $schema->getTestFrameworkExtraOptions() ?? '';
```

Comment:
Wouldn't it be better to force `getTestFrameworkExtraOptions()` to be a `string` here instead of `string|null`? That would avoid the necessity of the last `??`

Reply:
Also if it's a single line might be worth to just inline this method